crates/ruff_linter/resources/test/fixtures/pylint/stop_iteration_return.py

```diff
@@ -0,0 +1,129 @@
+"""Test cases for PLR1708 stop-iteration-return."""
+
+# Valid cases - should not trigger the rule
+def normal_function():
+    raise StopIteration  # Not a generator, should not trigger
+
+
+def normal_function_with_value():
+    raise StopIteration("value")  # Not a generator, should not trigger
+
+
+def generator_with_return():
+    yield 1
+    yield 2
+    return "finished"  # This is the correct way
+
+
+def generator_with_yield_from():
+    yield from [1, 2, 3]
+
+
+def generator_without_stop_iteration():
+    yield 1
+    yield 2
+    # No explicit termination
+
+
+def generator_with_other_exception():
+    yield 1
+    raise ValueError("something else")  # Different exception
+
+
+# Invalid cases - should trigger the rule
+def generator_with_stop_iteration():
+    yield 1
+    yield 2
+    raise StopIteration  # Should trigger
+
+
+def generator_with_stop_iteration_value():
+    yield 1
+    yield 2
+    raise StopIteration("finished")  # Should trigger
+
+
+def generator_with_stop_iteration_expr():
+    yield 1
+    yield 2
+    raise StopIteration(1 + 2)  # Should trigger
+
+
+def async_generator_with_stop_iteration():
+    yield 1
+    yield 2
+    raise StopIteration("async")  # Should trigger
+
+
+def nested_generator():
+    def inner_gen():
+        yield 1
+        raise StopIteration("inner")  # Should trigger
+    
+    yield from inner_gen()
+
+
+def generator_in_class():
+    class MyClass:
+        def generator_method(self):
+            yield 1
+            raise StopIteration("method")  # Should trigger
+    
```

Comment:
There's also some extra indentation and whitespace in here. I suggest running `ruff format` on this fixture and then re-running the tests and accept the snapshot changes.